internals/cli/env.go

```diff
@@ -165,47 +158,211 @@ func (a *App) CheckStrictEnv() error {
+	} else {
+		cmd.Flags().BoolVar(reference, name, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) IntVar(reference *int, name string, def int, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().IntVar(reference, name, def, usage)
+	} else {
+		cmd.Flags().IntVar(reference, name, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) StringVar(reference *string, name string, def string, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().StringVar(reference, name, def, usage)
+	} else {
+		cmd.Flags().StringVar(reference, name, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) DurationVar(reference *time.Duration, name string, def time.Duration, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().DurationVar(reference, name, def, usage)
+	} else {
+		cmd.Flags().DurationVar(reference, name, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) VarP(reference pflag.Value, name string, shorthand string, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().VarP(reference, name, shorthand, usage)
+	} else {
+		cmd.Flags().VarP(reference, name, shorthand, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) Var(reference pflag.Value, name string, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().Var(reference, name, usage)
+	} else {
+		cmd.Flags().Var(reference, name, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) VarPF(reference pflag.Value, name string, shorthand string, usage string, hasEnv bool, persistent bool) *pflag.Flag {
+	var flag *pflag.Flag
+	if persistent {
+		flag = cmd.PersistentFlags().VarPF(reference, name, shorthand, usage)
+	} else {
+		flag = cmd.Flags().VarPF(reference, name, shorthand, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+	return flag
 }
 
 // Command adds a new subcommand to this command.
-func (cmd *CommandClause) Command(name, help string) *CommandClause {
-	return &CommandClause{
-		CmdClause: cmd.CmdClause.Command(name, help),
-		name:      name,
-		app:       cmd.app,
+func (cmd *CommandClause) CreateCommand(name, help string) *CommandClause {
+	clause := &CommandClause{
+		Command: func() *cobra.Command {
+			newCommand := &cobra.Command{Use: name, Short: help}
+			return newCommand
+		}(),
+		name: name,
+		App:  cmd.App,
 	}
+	cmd.AddCommand(clause.Command)
+	return clause
 }
 
 // Hidden hides the command in help texts.
 func (cmd *CommandClause) Hidden() *CommandClause {
-	cmd.CmdClause = cmd.CmdClause.Hidden()
+	cmd.Command.Hidden = true
 	return cmd
 }
 
+func (cmd *CommandClause) FullCommand() string {
```

Comment:
I think this method should not be exported (unless necessary).

Reply:
Since it is only used in one place (which is in the same file), we can fix it

Reply:
yes, fixed it